src/Libraries/Microsoft.Extensions.DependencyInjection.AutoActivation/AutoActivationExtensions.Keyed.cs

```diff
@@ -85,6 +81,10 @@ public static partial class AutoActivationExtensions
                 }
 
                 _ = ao.KeyedAutoActivators.Add((serviceType, serviceKey));
+
+                [UnconditionalSuppressMessage("AotAnalysis", "IL3050:RequiresDynamicCode",
+                    Justification = "When IsDynamicCodeSupported is not supported, DependencyInjection ensures IEnumerable service types are not a ValueType.")]
```

Comment:
Just curious, Is this false positive something that can be fixed in the analyzer at some point?

Reply:
Unfortunately, no. The static analysis isn't able to know that this `IEnumerable<T>` is only going to be used in DependencyInjection, and it is only ever going to be ref types. So this is a case where we have to suppress the warning since we have more context than the tools can know.